internal/onepassword/model/item.go

```diff
@@ -329,3 +369,190 @@ func buildSectionMap(item *sdk.Item) map[string]ItemSection {
+		})
+	}
+	return modelURLs
+}
+
+func fromConnectSections(sections []*connect.ItemSection, sectionMap map[string]ItemSection) []ItemSection {
+	modelSections := make([]ItemSection, 0, len(sections))
+	for _, s := range sections {
+		if s != nil {
+			section := ItemSection{
+				ID:    s.ID,
+				Label: s.Label,
+			}
+			modelSections = append(modelSections, section)
+			sectionMap[s.ID] = section
+		}
+	}
+	return modelSections
+}
+
+func fromConnectFields(fields []*connect.ItemField, sectionMap map[string]ItemSection) []ItemField {
+	modelFields := make([]ItemField, 0, len(fields))
+	for _, f := range fields {
+		if f == nil {
+			continue
+		}
+
+		field := ItemField{
+			ID:      f.ID,
+			Label:   f.Label,
+			Type:    ItemFieldType(f.Type),
+			Value:   f.Value,
+			Purpose: ItemFieldPurpose(f.Purpose),
+		}
+
+		// Associate field with section if applicable
+		if f.Section != nil && f.Section.ID != "" {
+			if section, exists := sectionMap[f.Section.ID]; exists {
+				field.SectionID = section.ID
+				field.SectionLabel = section.Label
+			}
+		}
+
+		// Handle password recipe if present
+		if f.Recipe != nil {
+			characterSets := make([]CharacterSet, len(f.Recipe.CharacterSets))
+			for i, cs := range f.Recipe.CharacterSets {
+				characterSets[i] = CharacterSet(cs)
+			}
+
+			field.Recipe = &GeneratorRecipe{
+				Length:        f.Recipe.Length,
+				CharacterSets: characterSets,
+			}
+		}
+
+		modelFields = append(modelFields, field)
+	}
+	return modelFields
+}
+
+func fromConnectFiles(files []*connect.File, sectionMap map[string]ItemSection) []ItemFile {
+	result := make([]ItemFile, 0, len(files))
+	for _, f := range files {
+		if f == nil {
+			continue
+		}
+
+		itemFile := ItemFile{
+			ID:          f.ID,
+			Name:        f.Name,
+			Size:        f.Size,
+			ContentPath: f.ContentPath,
+		}
+
+		// Only set Section if it exists
+		if f.Section != nil && f.Section.ID != "" {
+			if section, exists := sectionMap[f.Section.ID]; exists {
+				itemFile.SectionID = section.ID
+				itemFile.SectionLabel = section.Label
+			}
+		}
+
+		result = append(result, itemFile)
+	}
+	return result
+}
+
+func toConnectURLs(urls []ItemURL) []connect.ItemURL {
+	connectURLs := make([]connect.ItemURL, 0, len(urls))
+	for _, u := range urls {
+		connectURLs = append(connectURLs, connect.ItemURL{
+			URL:     u.URL,
+			Label:   u.Label,
+			Primary: u.Primary,
+		})
+	}
+	return connectURLs
+}
+
+func toConnectSections(sections []ItemSection) []*connect.ItemSection {
+	connectSections := make([]*connect.ItemSection, 0, len(sections))
+	for _, s := range sections {
+		connectSections = append(connectSections, &connect.ItemSection{
+			ID:    s.ID,
+			Label: s.Label,
+		})
+	}
+	return connectSections
+}
+
+func toConnectFields(fields []ItemField) []*connect.ItemField {
+	connectFields := make([]*connect.ItemField, 0, len(fields))
+	for _, f := range fields {
+		if f.Generate && f.Recipe != nil {
+			password, err := generatePassword(f.Recipe)
```

Comment:
Generate password uses `sdk` under the hood and will fail as user authenticated with Connect. It should just pass password recipe as a part of `connect.Item` in `connectClient.Create` request and Connect will generate the password for us.